src/intrinsics/atomic.rs

```diff
@@ -199,21 +202,32 @@ trait EvalContextPrivExt<'tcx>: MiriInterpCxExt<'tcx> {
             span_bug!(this.cur_span(), "atomic arithmetic operation type mismatch");
         }
 
+        // Inform GenMC about the atomic rmw operation.
+        if let Some(genmc_ctx) = this.machine.data_race.as_genmc_ref() {
+            let is_unsigned = matches!(rhs.layout.ty.kind(), ty::Uint(_));
+            let (old, _is_success) = genmc_ctx.atomic_rmw_op(
+                this,
```

Comment:
Why is this hooked here and not inside `atomic_min_max_scalar`/`atomic_rmw_op_immediate`?